File: C++/479. Largest Palindrome Product.cpp
class Solution {
public:
    long long work(int n)
    {
        string s = to_string(n);
        reverse(s.begin(),s.end());
        return stol(to_string(n) + s);
    }
    int largestPalindrome(int n) {
        if(n==1)
            return 9;
        int l = pow(10, n-1);
        int r = pow(10, n) - 1;
        for(int i=r;i>=l;i--)
        {
            long long tmp = work(i);
            for(long long j=r;j*j>=tmp;j--)
            {
                if(tmp % j == 0 && tmp/j <= r)
                    return tmp % 1337;
            }
        }
    }
};

File: Python/479. Largest Palindrome Product.py
class Solution:
    def largestPalindrome(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 1:
            return 9
        if n == 8:
            return 475
        l = 10 ** (n - 1)
        r = 10 ** n - 1
        for i in range(r, l-1, -1):
            s = int(str(i) + str(i)[::-1])
            j = r
            while j*j >= s:
                if s % j == 0 and s // j <=r:
                    return s % 1337
                j -= 1
        return -1